File: js/emit_reciprocal.js
function
emit_reciprocal(u, p) // p = y^x where x is a negative number
{
	var num, den, v;

	num = {type:LINE, a:[], level:u.level};
	den = {type:LINE, a:[], level:u.level};

	emit_roman_text(num, "1");

	if (isminusone(caddr(p)))
		emit_expr(den, cadr(p));
	else {
		emit_base(den, cadr(p));
		emit_numeric_exponent(den, caddr(p)); // sign is not emitted
	}

	if (num.a.length == 1)
		num = num.a[0];
	else
		emit_update(num);

	if (den.a.length == 1)
		den = den.a[0];
	else
		emit_update(den);

	v = {type:FRACTION, num:num, den:den, level:u.level};

	emit_update_fraction(v);

	u.a.push(v);
}

File: app/emit_reciprocal.c
#include "app.h"

// p = y^x where x is a negative number

void
emit_reciprocal(struct atom *p)
{
	int t;

	emit_roman_char('1'); // numerator

	t = tos;

	if (isminusone(caddr(p)))
		emit_expr(cadr(p));
	else {
		emit_base(cadr(p));
		emit_numeric_exponent(caddr(p)); // sign is not emitted
	}

	emit_update_list(t);

	emit_update_fraction();
}
